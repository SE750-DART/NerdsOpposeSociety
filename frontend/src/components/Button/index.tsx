import React from "react";
import styles from "./style.module.css";

interface ButtonProps {
  variant?: "primary" | "secondary";
  size?: "big" | "small";
  text: string;
  handleOnClick: React.MouseEventHandler;
  disabled?: boolean;
}

const Button = ({
  variant = "primary",
  size = "big",
  text,
  handleOnClick,
  disabled = false,
}: ButtonProps) => {
  let buttonTypeStyle: string = "";
  if (variant === "primary") {
    buttonTypeStyle = styles.button__primary;
  } else if (variant === "secondary") {
    buttonTypeStyle = styles.button__secondary;
  }

  let buttonSizeStyle: string = "";
  if (size === "big") {
    buttonSizeStyle = styles.button__big;
  } else if (size === "small") {
    buttonSizeStyle = styles.button__small;
  }

  let buttonDisabledStyle: string = "";
  if (disabled) {
    buttonDisabledStyle = styles.button__disabled;
  }

  return (
    <button
      className={`${styles.button} ${buttonTypeStyle} ${buttonSizeStyle} ${buttonDisabledStyle}`}
      type="button"
      onClick={handleOnClick}
      disabled={disabled}
    >
      {text}
    </button>
  );
};

export default Button;
