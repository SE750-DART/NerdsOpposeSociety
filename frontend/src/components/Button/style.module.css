.button {
  align-items: center;
  border: none;
  border-radius: 16px;
  cursor: pointer;
  display: flex;
  font-family: "DM Sans", sans-serif;
  font-weight: bold;
  justify-content: center;
  overflow: hidden;
  padding: 0 24px;
  text-overflow: ellipsis;
  transition: 0.3s;
  white-space: nowrap;
  width: fit-content;
}

.button:hover {
  transition: 0.3s;
}

.button__primary {
  background-color: #000000;
  box-shadow: 0px 10px 40px rgba(0, 0, 0, 0.5);
  color: #eeffff;
}

.button__primary:hover {
  box-shadow: 0px 10px 50px rgba(0, 0, 0, 0.6);
}

.button__secondary {
  background-color: #eeffff;
  box-shadow: 0px 10px 40px rgba(0, 0, 0, 0.25);
  color: #000000;
}

.button__secondary:hover {
  box-shadow: 0px 10px 50px rgba(0, 0, 0, 0.4);
}

.button__big {
  font-size: 32px;
  height: 64px;
  width: 100%;
}

.button__small {
  box-shadow: 0px 10px 20px rgba(0, 0, 0, 0.1);
  font-size: 24px;
  height: 48px;
}

.button__small:hover {
  box-shadow: 0px 10px 30px rgba(0, 0, 0, 0.25);
}

.button__disabled {
  background-color: #cfcfcf;
  box-shadow: none;
  color: #000000;
  cursor: not-allowed;
}

.button__disabled:hover {
  box-shadow: none;
}
