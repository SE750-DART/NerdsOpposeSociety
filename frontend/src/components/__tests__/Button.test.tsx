import React from "react";
import renderer from "react-test-renderer";
import Button from "../Button";

it("Button snapshot testing", () => {
  const treePrimary = renderer
    .create(<Button text="Start game" handleOnClick={() => {}} />)
    .toJSON();
  expect(treePrimary).toMatchSnapshot();

  const treeSecondary = renderer
    .create(
      <Button text="Start game" handleOnClick={() => {}} variant="secondary" />
    )
    .toJSON();
  expect(treeSecondary).toMatchSnapshot();
});
