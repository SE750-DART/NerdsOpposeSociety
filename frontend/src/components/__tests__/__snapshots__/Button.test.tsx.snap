// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot testing 1`] = `
<button
  className="button button__primary button__big "
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Start game
</button>
`;

exports[`Button snapshot testing 2`] = `
<button
  className="button button__secondary button__big "
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Start game
</button>
`;
